src/App.tsx

```diff
@@ -50,22 +51,117 @@ import ContentManager from "./api/ContentManager"
 import { Plugins, AppState } from '@capacitor/core';
 import NotificationsAssistant from "./api/NotificationsAssistant"
 
+
+
 // init content manager
 let cm = new ContentManager();
 
 const App: React.FC = () => {
 
+
   // capacitor notifications setup
-  useIonViewDidEnter(() => {
+  useIonViewWillEnter(() => {
     // refresh notifications on app enter
-    Plugins.App.addListener("appStateChange", (state: AppState) => {
+   
+  });
+
+  useEffect( () => {
+
+    //console.log("Annyeonghaseyo")
+
+    let notificationAssistant = new NotificationsAssistant();
+
+    var remain:Boolean;
+    var starting:Number;
+    var ending:Number;
+
+    notificationAssistant.NoNotificationsRemaining().then((remaining) => {
+      if(!remaining){
+        console.log("There are notifications that are already scheduled");
+      }else{
+        console.log("There are notifications that aren't scheduled")
+      }
+      remain = remaining;
+    } ).then(()=>
+      notificationAssistant.GetStart()
+    ).then((start)=>{
+      starting = start
+    }).then(()=>notificationAssistant.GetEnd()).then((end)=>{
+      ending = end
+    }).then(()=>{
+
+      var dateToday = new Date()
+
+      var hour = dateToday.getHours()
+      var minutes = dateToday.getMinutes()
+
+      var militaryTime:Number = hour*100 + minutes
+
+      if(remain && (militaryTime < starting && militaryTime < ending)){
+        notificationAssistant.NotificationSetter();
+      }
+
+    })
+  })
+
+   console.log("Starting app")
+   //Seems to not be called during startup
+   //https://stackoverflow.com/questions/50623279/js-event-handler-async-function/50623441
+   Plugins.App.addListener("appStateChange", async (state: AppState) => {
       if (state.isActive) {
         // call the notification setup
-        let notificationAssistant = new NotificationsAssistant();
+        console.log("App is now active")
+
+         let notificationAssistant = new NotificationsAssistant();
+
+         var remaining = await notificationAssistant.NoNotificationsRemaining()
+         console.log(remaining)
+         if(!remaining){
+           console.log("There are notifications that are already scheduled");
+           return;
+         }else{
+
+           //check to see time 
+           var dateToday = new Date()
+
+           var hour = dateToday.getHours()
+           var minutes = dateToday.getMinutes()
+
+           var militaryTime:Number = hour*100 + minutes
+
+           var start = await notificationAssistant.GetStart()
+           var end = await notificationAssistant.GetEnd()
+
+            //probably only do it for the end
+           if(militaryTime >= start || militaryTime >= end){
```

Comment:
Again... This code should be scheduling notifications regardless of what time of the day the app is opened.... Even if you shouldn't receive notifications at a given time, you can still schedule notifications at that time for later. Currently the code doesn't even try to schedule the notifications for later. The logic to schedule notifications within valid time range going into the future should be in the notificationsAssistant.